ui/src/app/pages/admin/admin-institution-edit.tsx

```diff
@@ -154,44 +165,72 @@ export const AdminInstitutionEdit = withUrlParams()(class extends React.Componen
-    this.setState(fp.set(['institution', attribute], emailList.map(email => email.trim())));
+    // For now, only RT requirement is supported, so fine to set tierEmailAddresses to an single element array.
+    this.setState(fp.set(['institution', attribute],
+        [{accessTierShortName: AccessTierShortNames.Registered, emailAddresses: emailList.map(email => email.trim())}]));
+  }
+
+  setDomains(emailInput, attribute) {
+    const emailList = emailInput.split(/[,\n]+/);
+    // For now, only RT requirement is supported, so fine to set tierEmailAddresses to an single element array.
+    this.setState(fp.set(['institution', attribute],
+        [{accessTierShortName: AccessTierShortNames.Registered, emailDomains: emailList.map(email => email.trim())}]));
   }
 
   // Check if the fields have not been edited
-  fieldsNotEdited() {
+  fieldsEdited() {
     return (this.isAddInstitutionMode && !this.fieldsNotEditedAddInstitution)
         || (institutionToEdit && this.fieldsNotEditedEditInstitution);
   }
 
   get fieldsNotEditedAddInstitution() {
     const {institution} = this.state;
     return institution.displayName || institution.userInstructions ||
-      institution.organizationTypeEnum || institution.duaTypeEnum || institution.emailAddresses || institution.emailDomains;
+      institution.organizationTypeEnum || institution.tierRequirements
+        || institution.duaTypeEnum || institution.tierEmailDomains
+        || institution.tierEmailAddresses;
   }
 
   get fieldsNotEditedEditInstitution() {
     const {institution} = this.state;
     return institution.displayName === institutionToEdit.displayName &&
         institution.organizationTypeEnum === institutionToEdit.organizationTypeEnum &&
         institution.duaTypeEnum === institutionToEdit.duaTypeEnum &&
-        institution.emailAddresses === institutionToEdit.emailAddresses &&
-        institution.emailDomains === institutionToEdit.emailDomains &&
+        institution.tierEmailAddresses === institutionToEdit.tierEmailAddresses &&
+        institution.tierEmailDomains === institutionToEdit.tierEmailDomains &&
+        institution.tierRequirements === institutionToEdit.tierRequirements &&
         institution.userInstructions === institutionToEdit.userInstructions &&
         institution.organizationTypeOtherText === institutionToEdit.organizationTypeOtherText;
   }
 
-
-  validateRequiredFields() {
+  hasInvalidFields() {
     const {institution} = this.state;
-    let emailValid = false;
-    if (institution.duaTypeEnum) {
-      emailValid = institution.duaTypeEnum === DuaType.MASTER ?
-          institution.emailDomains !== undefined : institution.emailAddresses !== undefined;
+    let emailValid = true;
+    if (!institution.tierRequirements) {
+      // It is not expected for a tier requirement to be empty
+      return false;
+    }
+    const rtRequirement = getRegisteredTierRequirement(institution);
+    if (rtRequirement && rtRequirement.membershipRequirement
```

Comment:
nit: using a switch, or our functional version `switchCase` might be cleaner here